libnd4j/include/ops/declarable/impl/DeclarableOp.cpp

```diff
@@ -335,7 +347,7 @@ int sd::ops::DeclarableOp::prepareOutputs(Context &ctx) {
           int shapeEquals = shape::equalsSoft(out, array->shapeInfo());
           int arrayEmpty = array->isEmpty();
           // checking out shape equality
-          if (!shapeEquals || arrayEmpty) {
+          if (!shapeEquals) {
```

Comment:
was ommiting "arrayEmpty check" here deliberate?

Reply:
Yes. It was causing an error to be thrown when one didn't need to be. This came up in quite a few of the TF import tests. Shapes would be equal but then it would complain about something being invalid. Just because it's empty doesn't mean it's invalid.